Src/Client.ts

```diff
@@ -224,82 +225,94 @@ export class Client {
         req.setJobUuid(jobUuid);
         req.setToken(this.token);
 
-        let isOk = true;
-        let message: string = "ok";
-        let statuses: JobStatus[] = [];
-        let results: string[][] | string[][][] = [];
-        let secrets: any = [];
-
         const promises = [];
-        for(let i = 0; i < this.clients.length; i++) {
-            promises.push(this.clients[i].getComputationResult(req, {}))
+        for (const client of this.clients) {
+            const call = client.getComputationResult(req, {});
+
+            // Receive data via stream
+            const resList: {isOk: boolean, pieceId: number, result: string, status: JobStatus}[] = [];
+            call.on('data', function(response: any) {
+                resList.push({
+                  isOk: response.getIsOk(),
+                  pieceId: response.getPieceId(),
+                  result: response.getResult(),
+                  status: response.getStatus()
+                });
+            });
+
+            promises.push(new Promise((resolve, reject) => {
+                call.on('end', function() {
+                    // concatenate results in pieceId order
+                    resList.sort((a, b):number => a.pieceId - b.pieceId);
+                    let result = "";
+                    let isOk = true;
+                    for (const res of resList) {
+                        result += JSON.parse(res.result);
```

Comment:
[MEMO] My understanding here is that JSON.parse is used to parse the string.

example:
```
res.result: '"\"[0,"'
JSON.parse(res.result): '"[0.'
```

Reply:
[MEMO] In TypeScript (JavaScript), `+=` operation is not problem for performance to concat string.

Reply:
I will resolve it.